lib/src/utils.dart

```diff
@@ -275,12 +275,14 @@ class Utils {
     required List<VideoParameters> presets,
   }) {
     List<rtc.RTCRtpEncoding> result = [];
+    final rids =
+        videoRids.sublist(videoRids.length - presets.length, videoRids.length);
```

Comment:
sorry I wasn't following this algo? In JS I believe we are just looking for the first N presets that match the rids. so if I gave two presets [low, med], it should map to `low: q, med: h`

with this algo, it seems that it would map to `low: h, med: f`. also if I give it 4 presets, this would crash? (i.e. `sublist(-1, 3)`

Reply:
My understanding is that if there are only two layers then it is full and half, f,h
but if the rid just marks the track layer and is not used to distinguish layer between full, half, and quarter, then I can revert the changes here

Reply:
it should be q, h if there are only two layers. I think the server side expects it to go from q to f

also the crash seems like it'd happen regardless?

Reply:
set to h, f seems to get warnings, https://github.com/livekit/livekit/blob/master/pkg/sfu/buffer/videolayerutils.go#L75-L93 but it should not crash. I tested before reverting, and the livekit server did not crash. I have reverted and followed the server settings and no exceptions will be thrown.